include/oneapi/tbb/task_group.h

```diff
@@ -51,7 +51,7 @@ class task_group_base;
 namespace r1 {
 // Forward declarations
 class tbb_exception_ptr;
-class market;
+class cancellation_disseminator;
```

Comment:
I mean using the generalization of a particular pieces of data the class store.
`car_factory` is a good name because it says that the class with that name manages all the logic and states related to manufacturing of cars.

I would expect to name it something like `cancellation_thread_list` or similarly, with the accent on the list of threads since it is what this class actually maintains. Perhaps, even move the word "cancellation" in the corresponding method name, with possible extension of the method list so that the class does not only cancellation propagation.

Reply:
The real propose of this class is to collect inside it external and worker threads. It doesn't matter how they are stored inside the class. The name `cancellation_disseminator` here have similar properties with `car_factry` since it manage cancelation logic across TBB via storing all types of threads and disseminate the signal across them - this is a abstraction for this class.